src/libAtomVM/term.h

```diff
@@ -1001,7 +1001,9 @@ static inline term term_create_uninitialized_binary(uint32_t size, Heap *heap, G
 static inline term term_from_literal_binary(const void *data, uint32_t size, Heap *heap, GlobalContext *glb)
 {
     term binary = term_create_uninitialized_binary(size, heap, glb);
-    memcpy((void *) term_binary_data(binary), data, size);
+    if (size > 0) {
+        memcpy((void *) term_binary_data(binary), data, size);
```

Comment:
do we really need this specific check? I think that `memcpy` already deals with 0 bytes size.

Reply:
from C99 standard:
```
Where an argument declared as size_t n specifies the length of the array for a function, n can have the value zero on a call to that function. Unless explicitly stated otherwise in the description of a particular function in this subclause, pointer arguments on such a call shall still have valid values, as described in 7.1.4. On such a call, a function that locates a character finds no occurrence, a function that compares two character sequences returns zero, and a function that copies characters copies zero characters.
```
so `size == 0` is ok, while `src` and `dst` must be always valid pointers.

Reply:
I vote no, it is an implicit contract that the ptr passed to `term_from_literal_binary` is valid, as it always has been.  So no change there.  This (corrected) PR now obeys this contract.